site-scrapers/BaystateHealth/index.js

```diff
@@ -0,0 +1,55 @@
+const { site } = require("./config");
+
+/*
+ * This function calls ScrapeWebsiteData to gather availability data from the
+ * site
+ *
+ * @param browser created when Puppeteer connects to a Chromium instance
+ * @return JSON blob with { hasAppointments: Bool, totalAvailability: Int }
+ */
+module.exports = async function GetAvailableAppointments(browser) {
+    if (browser.browserContexts().length > 1) {
+        await console.log(`${site.name} testsuite started...`);
+        const webData = await ScrapeWebsiteData(browser);
+        await console.log(`${site.name} testsuite finished...`);
+        return {
+            ...site,
+            ...webData,
+            timestamp: new Date(),
+        };
+    } else {
+        await console.log(`${site.name} starting.`);
+        const page = await browser.newPage();
+        await page.goto(site.signUpLink);
+        const webData = await ScrapeWebsiteData(browser);
+        await console.log(`${site.name} done.`);
+        return {
+            ...site,
+            ...webData,
+            timestamp: new Date(),
+        };
+    }
+};
+
+async function ScrapeWebsiteData(browser) {
+    let pageArray = await browser.pages();
+    const page = pageArray[1];
```

Comment:
I don't think we can count on this. When the scrapers run they open up a LOT of pages. You should just move lines 22 and 23 into ScrapeWebsiteData, or pass the page through to that function.